Prevent root_dir option from being empty

lime opened the pull request:
This is a quick fix for #340.

When setting root_dir to an empty string (as is done in the default rake task config) this line previously caused root_dir to become an empty array.

This in turn caused factories, specs etc not to be annotated.

This is just a quick band-aid on a larger problem. We have option parsing spread out over many different places, with slight mismatches like this in the assumptions made.

lime commented:
I would probably approach the underlying issue by creating a `Annotate::Options` class, which could encapsulate all the quirks of our available options.

Then each place that uses these options (`bin/annotate`, rake tasks, direct invocation from code) wouldn't have to encode the same fallbacks & defaults, reducing the risk for bugs like this one.

I most likely won't have time to attempt that kind of overhaul, but maybe someone else would like to give it a try?

ctran commented on lib/tasks/annotate_models.rake:
```diff
@@ -21,7 +21,7 @@ task annotate_models: :environment do
   options[:simple_indexes] = Annotate.true?(ENV['simple_indexes'])
   options[:model_dir] = ENV['model_dir'] ? ENV['model_dir'].split(',') : ['app/models']
-  options[:root_dir] = ENV['root_dir'] ? ENV['root_dir'].split(',') : ['']
+  options[:root_dir] = ENV['root_dir'] && !ENV['root_dir'].empty? ? ENV['root_dir'].split(',') : ['']
```
I wonder if we should fix it where options[:root_dir] being used, make it work whether it's [], [''], or ['', '',...].

lime replied:
100% yes, we should. :)

I wasn't sure at first of how large the surface of `root_dir` was, so I wanted to start off with this simple fix. It seems though like this could really easily be taken care of inside `AnnotateModels.root_dir`.

Would you like me to give that a shot?

ctran replied:
Please and thank you.

lime commented:
There, I changed the `root_dir` method to return `['']` for all blank values, split comma-separated strings and leave other values (i.e. arrays) untouched. How does that look?

lime commented:
> However, can I request that you move this logic into the setter (writer) so it's done once?

I guess, although personally I think I would keep it in the getter. :)

Should `model_dir` be changed then too, to keep some consistency between the two?

lime commented:
As an aside, how do you feel about the idea I mentioned earlier about moving all of this logic into `Annotate::Options` or something similar? If we know you are interested it might be more likely that someone takes a stab at it. :)

lime commented:
> However, can I request that you move this logic into the setter (writer) so it's done once?

I went to do this and then realized why it's problematic: with the getter you'll get `['']` even when you never set the value, while with the setter you run the risk of getting `nil` by default.

I would suggest we keep the current approach.
